src/driver.rs

```diff
@@ -410,6 +410,90 @@ impl DriverManager {
         Ok(Driver { c_driver })
     }
 
+    /// Get the Driver based on the file extension from filename
+    ///
+    /// Searches for the available extensions in the registered
+    /// drivers and returns the matches. The determined driver is
+    /// checked for writing capabilities as
+    /// [`Dataset::open`](Dataset::open) can already open datasets
+    /// with just path.
+    ///
+    /// See also: [`get_driver_by_name`](Self::get_driver_by_name)
+    ///
+    /// # Example
+    ///
+    /// ```rust, no_run
+    /// use gdal::DriverManager;
+    /// # fn main() -> gdal::errors::Result<()> {
+    /// let compatible_drivers =
+    ///     DriverManager::guess_drivers_for_write("test.gpkg", true)
+    ///         .iter()
+    ///         .map(|d| d.short_name())
+    ///         .collect::<Vec<String>>();
+    /// println!("{:?}", compatible_drivers);
+    /// # Ok(())
+    /// # }
+    /// ```
+    /// ```text
+    /// ["GPKG"]
+    /// ```
+    pub fn guess_drivers_for_write(filename: &str, is_vector: bool) -> Vec<Driver> {
```

Comment:
Sorry to bikeshed here, but should we consider:

1. Also providing `guess_drivers_for_read`
2. Have the signature `guess_drivers_for_write<P: AsRef<Path>>(filename: P, ...)` to be more coherent with other methods accepting files. (`&str` is `AsRef<Path>`).

Reply:
Not sure about `guess_drivers_for_read`. There's already an API that works for existing files, and do we really need to find a driver that doesn't exist?

Reply:
I did consider using `AsRef<Path>`, but I thought the connections like `PG:https://xxxxxx` might not work. And we need to convert it back to string to use the "endswith" or "split" for double extensions. Hence, I went with string. I haven't tested if those connections work as path or not though.